pallets/inactivity-tracking/src/lib.rs

```diff
@@ -600,11 +633,21 @@ impl<T: Config> AuthorNotingHook<Collator<T>> for Pallet<T> {
             <CurrentActivityTrackingStatus<T>>::get()
         {
             if start <= T::CurrentSessionIndex::session_index() {
-                for author_info in info {
-                    total_weight
-                        .saturating_accrue(Self::on_author_noted(author_info.author.clone()));
-                    total_weight.saturating_accrue(Self::on_chain_noted(author_info.para_id));
+                let mut authors = BTreeSet::new();
+                let mut chains = BTreeSet::new();
+
+                for AuthorNotingInfo {
+                    author,
+                    para_id,
+                    block_number: _,
+                } in info
+                {
+                    authors.insert(author.clone());
+                    chains.insert(*para_id);
```

Comment:
Maybe you can do it in one line using https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.unzip 

But if not its fine, should be identical.

Reply:
https://github.com/moondance-labs/tanssi/commit/05f7de88b700f98422dbffb550b83ee996e4dc73